src/hotspot/share/opto/runtime.cpp

```diff
@@ -1955,6 +2111,74 @@ NamedCounter* OptoRuntime::new_named_counter(JVMState* youngest_jvms, NamedCount
   return c;
 }
 
+void OptoRuntime::initialize_types() {
```

Comment:
For non-C2 runtime stubs, please, turn factories into local static functions and directly initialize cache fields in `OptoRuntime::initialize_types()`.

As an example: https://github.com/openjdk/jdk/commit/c343b56249f223a72cef4671cd5de15406387e3d